llama_index/indices/struct_store/sql_query.py

```diff
@@ -203,6 +206,150 @@ async def _aquery(self, query_bundle: QueryBundle) -> Response:
+                else:
+                    raise ValueError(f"Unknown table type: {table}")
+                table_info = self._sql_database.get_single_table_info(table_str)
+                context_strs.append(table_info)
+
+        else:
+            # get all tables
+            table_names = self._sql_database.get_table_names()
+            for table_name in table_names:
+                table_info = self._sql_database.get_single_table_info(table_name)
+                context_strs.append(table_info)
+
+        tables_desc_str = "\n\n".join(context_strs)
+        return tables_desc_str
+
+    @llm_token_counter("query")
+    def _query(self, query_bundle: QueryBundle) -> Response:
+        """Answer a query."""
+        table_desc_str = self._get_table_context(query_bundle)
+        logger.info(f"> Table desc str: {table_desc_str}")
+
+        response_str, _ = self._service_context.llm_predictor.predict(
+            self._text_to_sql_prompt,
+            query_str=query_bundle.query_str,
+            schema=table_desc_str,
+            dialect=self._sql_database.dialect,
+        )
+
+        sql_query_str = self._parse_response_to_sql(response_str)
+        # assume that it's a valid SQL query
+        logger.debug(f"> Predicted SQL query: {sql_query_str}")
+
+        raw_response_str, extra_info = self._sql_database.run_sql(sql_query_str)
+        extra_info["sql_query"] = sql_query_str
+
+        if self._synthesize_response:
+            response_str, _ = self._service_context.llm_predictor.predict(
+                self._response_synthesis_prompt,
+                query_str=query_bundle.query_str,
+                sql_query=sql_query_str,
+                sql_response_str=raw_response_str,
+            )
+        else:
+            response_str = raw_response_str
+
+        response = Response(response=response_str, extra_info=extra_info)
+        return response
+
+    @llm_token_counter("aquery")
+    async def _aquery(self, query_bundle: QueryBundle) -> Response:
+        """Answer a query."""
+        table_desc_str = self._get_table_context(query_bundle)
+        logger.info(f"> Table desc str: {table_desc_str}")
+
+        (
+            response_str,
+            formatted_prompt,
+        ) = await self._service_context.llm_predictor.apredict(
+            self._text_to_sql_prompt,
+            query_str=query_bundle.query_str,
+            schema=table_desc_str,
+            dialect=self._sql_database.dialect,
+        )
+
+        sql_query_str = self._parse_response_to_sql(response_str)
+        # assume that it's a valid SQL query
+        logger.debug(f"> Predicted SQL query: {sql_query_str}")
+
+        response_str, extra_info = self._sql_database.run_sql(sql_query_str)
+        extra_info["sql_query"] = sql_query_str
+        response = Response(response=response_str, extra_info=extra_info)
+        return response
+
+
 # legacy
 GPTNLStructStoreQueryEngine = NLStructStoreQueryEngine
```

Comment:
might be good to mark `NLStructStoreQueryEngine` and `SQLStructStoreQueryEngine` as deprecated in comments/docstring